FreeRTOS/Demo/Common/Minimal/TimerDemo.c

```diff
@@ -241,17 +241,22 @@ BaseType_t xAreTimerDemoTasksStillRunning( TickType_t xCycleFrequency )
         xLastCycleFrequency = xCycleFrequency;
     }
 
-    /* Calculate the maximum number of times that it is permissible for this
-     * function to be called without ulLoopCounter being incremented.  This is
-     * necessary because the tests in this file block for extended periods, and the
-     * block period might be longer than the time between calls to this function. */
-    xMaxBlockTimeUsedByTheseTests = ( ( TickType_t ) configTIMER_QUEUE_LENGTH ) * xBasePeriod;
-    xLoopCounterIncrementTimeMax = ( xMaxBlockTimeUsedByTheseTests / xCycleFrequency ) + 1;
-
     /* If the demo task is still running then the loop counter is expected to
      * have incremented every xLoopCounterIncrementTimeMax calls. */
     if( ulLastLoopCounter == ulLoopCounter )
     {
+        /* Calculate the maximum number of times that it is permissible for this
+         * function to be called without ulLoopCounter being incremented.  This is
+         * necessary because the tests in this file block for extended periods, and the
+         * block period might be longer than the time between calls to this function. */
+        xMaxBlockTimeUsedByTheseTests = ( ( TickType_t ) configTIMER_QUEUE_LENGTH ) * xBasePeriod;
+        if( xCycleFrequency != ( TickType_t ) 0 )
+        {
+            xLoopCounterIncrementTimeMax = ( xMaxBlockTimeUsedByTheseTests / xCycleFrequency ) + 1;
+        } else {
+            xLoopCounterIncrementTimeMax = ( TickType_t ) 0;
+        }
```

Comment:
Would you be open to an assertion instead of this conditional? A cycle frequency of 0 doesn't make sense in the context of these tests.